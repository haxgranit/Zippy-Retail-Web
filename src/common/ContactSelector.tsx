import PropTypes from 'prop-types';
import { Button, FormControl } from 'react-bootstrap';
import {
  Dispatch,
  SetStateAction,
  useEffect,
  useState,
} from 'react';
import { Contact } from '../api';
import { ContactTypeEnum } from '../constants/enum/ContactTypeEnum';
import { formatContactName } from '../Helpers';

function ContactSelector({
  contactList,
  selectedContact,
  setSelectedContact,
}: {
  contactList: Array<Contact>;
  selectedContact: Contact;
  setSelectedContact: Dispatch<SetStateAction<Contact>>;
}) {
  const [contactType, setContactType] = useState<ContactTypeEnum>(ContactTypeEnum.PERSONAL);
  const [isProcessing, setIsProcessing] = useState<boolean>(true);
  const [filteredContactList, setFilteredContactList] = useState<Array<Contact>>(contactList);
  const [searchValue, setSearchValue] = useState<string>('');
  const [initialized, setInitialized] = useState<boolean>(false);
  const [showContactList, setShowContactList] = useState<boolean>(false);

  const filter = (value: string) => {
    setSearchValue(value);
    setFilteredContactList(contactList.filter((contactItem) => {
      const search = value.toLowerCase();
      return contactItem?.firstName?.toLowerCase().includes(search)
        || contactItem?.lastName?.toLowerCase().includes(search)
        || contactItem?.email?.toLowerCase().includes(search);
    }));
  };

  useEffect(() => {
    if (!initialized && contactList.length) {
      setInitialized(true);
      setIsProcessing(false);
      filter('');
    }
  });

  return (
    <>
      <div className="contact-selector">
        <div className="contact-identifier">
          <FormControl
            placeholder="Enter @ZippyUsername or Email"
            value={selectedContact?.email}
            onChange={() => false}
          />
          <Button
            className={`btn-select-contact btn-transparent${selectedContact?.email ? ' active' : ''}`}
            type="button"
            onClick={() => {
              if (selectedContact?.email) {
                setSelectedContact({
                  id: null,
                  email: '',
                  firstName: '',
                  lastName: '',
                  phone: '',
                } as unknown as Contact);
              } else {
                setShowContactList(true);
              }
            }}
          >
            <i className="zippy-cash-icon zc-add" />
          </Button>
        </div>

        <div className="contact-list-wrap">
          {showContactList
            && (
              <div className="contact-list">
                <div className="zippy-btn-group btn-group" role="group" aria-label="Zippy Cash">
                  <Button
                    className={contactType === ContactTypeEnum.PERSONAL ? 'active' : ''}
                    onClick={() => setContactType(ContactTypeEnum.PERSONAL)}
                    disabled={contactType === ContactTypeEnum.PERSONAL}
                  >
                    Personal
                  </Button>
                  <Button
                    className={contactType === ContactTypeEnum.BUSINESS ? 'active' : ''}
                    onClick={() => setContactType(ContactTypeEnum.BUSINESS)}
                    disabled={contactType === ContactTypeEnum.BUSINESS}
                  >
                    Business
                  </Button>
                </div>
                <div className="contact-search">
                  <i className="zippy-cash-icon zc-search" />
                  <FormControl
                    placeholder="Search..."
                    value={searchValue}
                    onChange={(evt) => filter(evt.target.value)}
                  />
                </div>
                <div className="contact-list-container">
                  {filteredContactList.length > 0
                  && (
                    <div className="personal-contacts">
                      <h4>Contacts</h4>
                      {filteredContactList?.map((item: Contact) => (
                        <div className={`contact-item${selectedContact.id === item.id ? ' selected' : ''}`} key={item.id}>
                          <div className="contact-name">
                            {formatContactName(item.firstName, item.lastName)}
                          </div>
                          <button
                            className="btn-select-contact btn-transparent"
                            type="button"
                            onClick={() => {
                              if (item.email) {
                                setSelectedContact(item);
                                setShowContactList(false);
                              }
                            }}
                          >
                            <i className="zippy-cash-icon zc-personal" />
                          </button>
                        </div>
                      ))}
                    </div>
                  )}
                  {!initialized && filteredContactList.length === 0 && searchValue === '' && (
                    <div>Loading...</div>
                  )}
                  {initialized && filteredContactList.length === 0 && searchValue !== '' && (
                    <div>No match</div>
                  )}
                  {initialized && filteredContactList.length === 0 && searchValue === '' && (
                    <div>No contacts</div>
                  )}
                </div>
                <div className="action">
                  <Button
                    className="zippy-btn"
                    disabled={isProcessing}
                    onClick={() => false}
                  >
                    {isProcessing && <div className="loading spinner-border" role="status" />}
                    Add Contact
                  </Button>
                </div>
              </div>
            )}
        </div>
      </div>
    </>
  );
}

ContactSelector.propTypes = {
  contactList: PropTypes.arrayOf(PropTypes.object),
  selectedContact: PropTypes.objectOf(PropTypes.any),
  setSelectedContact: PropTypes.func,
};

ContactSelector.defaultProps = {
  contactList: [],
  selectedContact: {
    id: null,
    email: '',
    firstName: '',
    lastName: '',
    phone: '',
  },
  setSelectedContact: undefined,
};

export default ContactSelector;
