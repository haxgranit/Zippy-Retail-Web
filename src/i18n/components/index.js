import LegalEnCAText from "./legal/LegalEnCAText";
import LegalEnUSText from "./legal/LegalEnUSText";
import LegalEsUSText from "./legal/LegalEsUSText";
import LegalFrCAText from "./legal/LegalFrCAText";

export {
    LegalEnCAText,
    LegalEnUSText,
    LegalEsUSText,
    LegalFrCAText
}
