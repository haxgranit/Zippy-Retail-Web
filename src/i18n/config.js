import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import { LegalEnCAText, LegalEnUSText, LegalEsUSText, LegalFrCAText } from './components';

i18n.use(initReactI18next).init({
  fallbackLng: 'en-US',
  lng: 'en-US',
  resources: {
    'en-CA': {
      translations: {
        ...require('./locales/en-CA/translations.json'),
        legal: {
          text: (
            <>
              <LegalEnCAText />
            </>
          )
        }
      }
    },
    'en-US': {
      translations: {
        ...require('./locales/en-US/translations.json'),
        legal: {
          text: (
            <>
              <LegalEnUSText />
            </>
          )
        }
      }
    },
    'es-US': {
      translations: {
        ...require('./locales/es-US/translations.json'),
        legal: {
          text: (
            <>
              <LegalEsUSText />
            </>
          )
        }
      }
    },
    'fr-CA': {
      translations: {
        ...require('./locales/fr-CA/translations.json'),
        legal: {
          text: (
            <>
              <LegalFrCAText />
            </>
          )
        }
      }
    },
  },
  returnObjects: false,
  returnedObjectHandler: (key, value, options) => value,
  ns: ['translations'],
  defaultNS: 'translations'
});

i18n.languages = ['en-CA', 'en-US', 'es-US', 'fr-CA'];

export default i18n;