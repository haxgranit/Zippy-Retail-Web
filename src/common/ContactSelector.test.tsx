import '../i18n/config';
import ContactSelector from './ContactSelector';

const ReactTestRenderer = require('react-test-renderer');

describe('ContactSelector Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <ContactSelector />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
