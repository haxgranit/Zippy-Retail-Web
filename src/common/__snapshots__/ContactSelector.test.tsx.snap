// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactSelector Component matches the snapshot 1`] = `
<div
  className="contact-selector"
>
  <div
    className="contact-identifier"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Enter @ZippyUsername or Email"
      value=""
    />
    <button
      className="btn-select-contact btn-transparent btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="zippy-cash-icon zc-add"
      />
    </button>
  </div>
  <div
    className="contact-list-wrap"
  />
</div>
`;
